contracts/Ethleverage.sol

```diff
@@ -0,0 +1,292 @@
+pragma solidity ^0.4.16;
+
+import "./SafeMath.sol";
+import "./Interface/ICDPContract.sol";
+import "./Interface/IDaiContract.sol";
+import "./Interface/IwethContract.sol";
+import "./Interface/IpethContract.sol";
+import "./Interface/IMKRContract.sol";
+import "./Interface/IMoneyMaker.sol";
+import "./Interface/ILiquidator.sol";
+
+
+//To-Do: be sure to complete the ETH_Address authorization
+contract Ethleverage {
+  using SafeMath for uint;
+
+  //events
+
+  event CDPOpened(bytes32 ID);
+  event PethLocked(bytes32 ID, uint recycledPeth);
+  event dai2Peth(uint recycledPeth);
+
+  event LogCDPAddressChanged(address oldAddress, address newAddress);
+  event LogDaiAddressChanged(address oldAddress, address newAddress);
+
+  //Variables
+  struct Investor {
+    uint layers;        // number of layers down
+    uint prinContr;     // principle contribution in Eth
+    uint CR;            // collatorization ratio
+    bytes32 cdps;     //array of CDPs (only one!)
```

Comment:
Maybe we should rename this to something like cdpi since now it's the id of a cdp